components/exercises/chapter-4-oscar/DeepComparison.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from 'react';
+import { checkEqual, lastWeekList, currentWeekList, WeekItem } from './Utils';
+
+/**
+ * This represents an array of arrays of WeekItem objects, essentially creating a two-dimensional array of WeekItem objects. Each element of the outer array is an array of WeekItem objects.
+ */
+type GroceryList = WeekItem[][];
+
+const newElement: WeekItem = {
+  name: '🍅 Tomatoes',
+  description: '1kg Fresh tomatoes',
+  price: 0.5,
+};
+
+/**
+ * Renders a component that compares two grocery lists and allows adding/removing items.
+ */
+export default function DeepComparison(): JSX.Element {
+  const [groceryList, setGroceryList] = useState<GroceryList>([
+    lastWeekList,
+    currentWeekList,
+  ]);
+
+  /**
+   * Handles the click event for adding a new element to the grocery list.
+   */
+  function handleClickAdd(): void {
+    setGroceryList((prevGroceryList) => {
+      const updatedGroceryList: GroceryList = [...prevGroceryList];
+      updatedGroceryList[1] = [...updatedGroceryList[1], newElement];
+      return updatedGroceryList;
+    });
+  }
+
+  /**
+   * Handles the click event for removing the last element of the grocery list.
+   */
+  function handleClickRemove(): void {
+    setGroceryList((prevGroceryList) => {
+      const updatedGroceryList: GroceryList = [...prevGroceryList];
+      updatedGroceryList[1] = updatedGroceryList[1].slice(0, -1);
+      return updatedGroceryList;
+    });
+  }
+
+  return (
+    <div className='flex flex-col'>
+      <div className='mt-2 flex flex-row'>
+        <div className='mx-[10px] my-auto flex w-[300px] flex-1 flex-col rounded-md border-solid p-[10px] shadow-lg'>
+          <strong>Grocery of last week</strong>
+
+          {groceryList &&
+            groceryList[0].map((item) => (
+              <ol key={item.name}>
```

Comment:
These items don't really have an order so I think [ul](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul) is more semantic than [ol](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ol)